### PR TITLE
Feature/register form

### PR DESCRIPTION
**Pull request by eliasriise**

### CODE REVIEW

**Comment by FredrikMorstad** [comment]
cleanup commit history

File: backend/Pipfile [hunk 1]
```diff
@@ -14,4 +14,4 @@ email-validator = "1.1.1"
 pyjwt = "2.1.0"
 
 [requires]
-python_version = "3.8.10"
+python_version = "3.9.7"
```
Comment (FredrikMorstad) [inline backend/Pipfile]: This pipfile is ment for the server, do not push local changes :+1:

**Comment by eliasriise** [comment]
How do

File: frontend/src/components/molecules/Navbar/Navbar.tsx [hunk 2]
```diff
@@ -18,14 +19,21 @@ const Navbar = () => {
...
+    <Flex h="7.5vh" w="100%" bg="darkgreen" justifyContent="flex-end">
       <Center mr=".75rem">
         {authenticated ? (
           <Button onClick={logout}> Logout </Button>
         ) : (
           <Button onClick={moveToLogin}> Login </Button>
         )}
       </Center>
+      <Center mr=".75rem">
+        {authenticated ? (
```
Comment (FredrikMorstad) [inline frontend/src/components/molecules/Navbar/Navbar.tsx]: conditional render is more suitable then ternary or maybe move it in the first ternary :thinking:

File: backend/Pipfile (hunk 1 above)
Comment (eliasriise) [reply backend/Pipfile]: Tard, æ spør jo kordan man fjerne de

File: nodemcu/modules/esp32/config.json [hunk 3]
```diff
@@ -11,7 +11,7 @@
         "on" : 1
     },
     "wifi" : {
-        "ssid" : "Network_name",
-        "pwd" : "Password"
+        "ssid" : "PlantBridge",
```
Comment (FredrikMorstad) [inline nodemcu/modules/esp32/config.json]: remove :fire:  don't push passwords

File: nodemcu/modules/esp32/test.lua [hunk 4]
```diff
@@ -0,0 +1,15 @@
+pin = 4
```
Comment (FredrikMorstad) [inline nodemcu/modules/esp32/test.lua]: add separate commits for mcu-files

File: frontend/src/components/molecules/Navbar/Navbar.tsx (hunk 2 above)
Comment (eliasriise) [reply frontend/src/components/molecules/Navbar/Navbar.tsx]: I will try to move this, hold up :smiley_cat:

File: frontend/src/api/requests.ts [hunk 5]
```diff
@@ -31,6 +31,7 @@ export const post = async <T>(url: string, data: any, auth = false) => {
       "Access-Control-Allow-Origin": "*",
     },
   });
+  console.log(baseUrl + url);
```
Comment (FredrikMorstad) [inline frontend/src/api/requests.ts]: :fire:

File: frontend/src/components/molecules/Navbar/Navbar.tsx (hunk 2 above)
Comment (eliasriise) [reply frontend/src/components/molecules/Navbar/Navbar.tsx]: I fixed it

Comment (eliasriise) [reply frontend/src/components/molecules/Navbar/Navbar.tsx]: :100:

File: frontend/src/components/molecules/Forms/RegisterForm.tsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+                    setError('Unexpected error, could\'t create user');
+            }
+            return;
+        }
+        try{
+            await login(email, password);
+            setAuthenticated(true);
+        }
+        catch(error){
+            alert(`User created, but could\'t log in, redirecting to start page`);
```
Comment (FredrikMorstad) [inline frontend/src/components/molecules/Forms/RegisterForm.tsx]: Personally i would avoid alert, use chakra toast

File: frontend/src/components/molecules/Forms/RegisterForm.tsx [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+        }
+        catch (error){
+            switch (error.statusCode){
+                case 400:
+                    setError(error.message['detail']);
+                    break;
+                case 409:
+                    setError(error.message['detail']);
+                    break;
+                case 422:
```
Comment (FredrikMorstad) [inline frontend/src/components/molecules/Forms/RegisterForm.tsx:65]: case 422 empty?

File: frontend/src/api/requests.ts (hunk 5 above)
Comment (eliasriise) [reply frontend/src/api/requests.ts]: ekke d der deg?

File: frontend/src/components/molecules/Forms/RegisterForm.tsx (hunk 7 above)
Comment (eliasriise) [reply frontend/src/components/molecules/Forms/RegisterForm.tsx:65]: no, it leads to default case, implying it is handled

File: frontend/src/components/molecules/Forms/RegisterForm.tsx (hunk 6 above)
Comment (eliasriise) [reply frontend/src/components/molecules/Forms/RegisterForm.tsx]: OK, i will fix this

File: frontend/src/api/requests.ts (hunk 5 above)
Comment (eliasriise) [reply frontend/src/api/requests.ts]: ok i fix

File: frontend/src/components/molecules/Forms/RegisterForm.tsx (hunk 6 above)
Comment (eliasriise) [reply frontend/src/components/molecules/Forms/RegisterForm.tsx]: i fixed it

**Comment by FredrikMorstad** [comment]
LGTM!

**Pull request merged**